azavea/fb-gender-survey-dashboard
# Handle unavailable years for country

## Overview

Some geography/year combinations are unavailable. In those cases,
we indicate which years are available for the geography on both the
geography selection page and the years selection page, and prevent
users from selecting a year if no selected geographies have data
available for it.

If a user selects both available geographies for a year and an unavailable 
geography for a year, the unavailable geography is filtered from the chart 
data. This prevents the rendering of null values.

A bug in Chakra-UI was preventing the radio buttons in the year selector
from displaying an externally controlled value correctly. It has been fixed 
in the latest version, so ChakraUI has been upgraded.

Connects #123 

### Demo

Geography Selection
<img width="350" alt="Screen Shot 2021-11-03 at 3 10 42 PM" src="https://user-images.githubusercontent.com/21046714/140192406-05467e45-ebc7-4d4a-9dab-1dca47428db0.png">

Year Selection (partial availability)
<img width="1056" alt="Screen Shot 2021-11-03 at 3 10 49 PM" src="https://user-images.githubusercontent.com/21046714/140192440-54a3654c-76ef-4943-8e08-b90dc2149f01.png">
Year Selection (no availability)
<img width="463" alt="Screen Shot 2021-11-03 at 3 12 04 PM" src="https://user-images.githubusercontent.com/21046714/140192466-9d981176-9e8b-45f2-a088-6a28c12cead9.png">
Year Selection (complete availability)
<img width="441" alt="Screen Shot 2021-11-03 at 3 12 14 PM" src="https://user-images.githubusercontent.com/21046714/140192484-e67ff701-4024-41d6-8019-7b996143e7aa.png">

## Testing Instructions

 * In the Netlify preview, select a geography that is available for 2020 and 2021, then view the years selector. Both years should be available and neither should have an availability warning.
 * Select a geography that is available for only 2020. On the years page, 2021 should be disabled and show an availability warning. 
 * Select a geography that is available for only 2020 and a geography that is available for 2020 and 2021. On the years page, 2021 should show an availability warning. Select 2021.
 * On the charts page, the unavailable geography should not appear on any of the charts. The available geography should render as usual.
